index.js

```diff
@@ -269,10 +269,33 @@ function validateEmail(input, invalidLowercase) {
   return showError(input, invalidLowercase);
 }
 
+// local storage
+function saveUserDetails() {
+  const formData = new FormData(form);
+  const fullname = formData.get('fullname');
+  const email = formData.get('email');
+  const feedback = formData.get('feedback');
+  const myFormData = { name: fullname, email, feedback };
+  localStorage.setItem('myFormData', JSON.stringify(myFormData));
+}
+
 form.addEventListener('submit', (e) => {
   e.preventDefault();
   const emailValid = validateEmail(emailInput, INPUT_LOWERCASE);
   if (emailValid) {
     form.submit();
+    saveUserDetails();
```

Comment:
According to the project requirements:

![image](https://user-images.githubusercontent.com/57421823/157619162-df710192-3484-4a12-ae67-a6a09e6afb48.png)

Therefore:

- [ ]  Please make it so the data is saved when the user changes the content, even if there is no submission, and ensure the localStorage object is actually created.

Look:

![image](https://user-images.githubusercontent.com/57421823/157620832-6e9eba1b-b41b-4542-8277-1ed2d6534910.png)




## Explanation

This event listener only saves the data to local storage when the user submits the form.  This is the reason why the `userDetails `object is never created if it doesn't exist.  

I'm inferring that you don't have your problem on your end, because your browser already has the object saved.  Try deleting the localStorage so it is created again without submitting the form to properly debug this issue.

## HInts

I would just use `onchange=""` with your function `saveUserDetails` so you use the function that you already built!

That way, the user won't have to rewrite the data every time he/she visits the page. 💪